### ISSUE TITLE
'(react/prop-types): Missing in props validation' when props interface extends from another

### ISSUE THREAD

**Issue by ggd543** [yannickcr/eslint-plugin-react#2654]
code snippet
```tsx
import React, { SFC } from 'react';

const mapDispatch = (dispatch: storeDispatch) => ({
   removeCollection: dispatch.infoLibTable.removeCollection,
});

interface InfoLibTableProps extends ReturnType<typeof mapDispatch> {
}

const App: SFC<InfoLibTableProps> = (props) => {
   props.removeCollection(); 
}

```

.eslintrc 
```json
{ 
  "plugins": [ "eslint-plugin-react"]
}
```

when running eslint , I got the following warning: 
` warning  'removeCollection' is missing in props validation                                 react/prop-types
`

**Comment by srittau** [comment]
I also get this error for `x` with the following code since today's run of "yarn upgrade":

```typescript
interface Foo {
    x: number;
}

interface Bar extends Foo {
    y: string;
}

const Baz = ({ x, y }: Bar) => (
    <span>
        {x}
        {y}
    </span>
);
```

While yarn upgrade also upgraded `eslint-plugin-react` from 7.20.0 to 7.20.1, this doesn't seem to be the root cause if this error. Manually downgrading it to 7.20.0 or 7.19.0 still gives this error. I suspect it is due to one of `eslint-plugin-react`'s dependencies.

**Comment by srittau** [comment]
This error actually occurs a second time in our code base after the upgrade:

```typescript
function useBar(): { x: number; y: string } {
    return { x: 123, y: "" };
}

const Foo = () => {
    const props = useBar();
    return <span>{props.x}</span>; // 'x' is missing in props validation 
};
```

Renaming `props` to something more sensible fixes this, but maybe it's a hint for what's going on.

**Comment by jacogr** [comment]
So worked around all these changing all my extends. Now on the latest version now seems complains when the props are defined but the component returns `React.ReactElement<Props> | null` - with the `| null` it just breaks all over again. and example -

```js
interface Props {
  value?: string;
}

// without the | null, all ok, with it, it is broken
function Test ({ value }: Props): React.ReactElement<Props> | null {
  if (!value) {
    return null;
  }

  return <div>{value}</div>;
}
```

**Comment by einsiol** [comment]
I'm also having this problem, would be great to have this supported. My issue is as follows:

```typescript
type Props = {
    title: string;
    name?: string;
};
const Component: React.FC<Props> = ({title, name}) => (...)

```

This will give me the error that the props validation is missing

> '...' is missing in props validation **eslint(react/prop-types)**

**Comment by drodriguln** [comment]
> I'm also having this problem using `React.FunctionalComponent`, would be great to have this supported. My issue is as follows:
> 
> ```ts
> type Props = {
>     title: string;
>     name?: string;
> };
> const Component: React.FC<Props> = ({title, name}) => (...)
> ```
> 
> This will give me the error that the props validation is missing
> 
> > '...' is missing in props validation **eslint(react/prop-types)**

I had the same problem. Turns out it didn't like me returning a ternary. 🤷‍♂️

**Comment by aplassen** [comment]
@drodriguln, same thing for me 🤔

**Comment by aplassen** [comment]
Also, in other cases, I also get the error described in this issue related to extended props.

**Comment by goooseman** [comment]
By downgrading back to 7.20.0 (without downgrading any of libraries deps) the problem is resolved, so I believe the problem appeared somewhere between 7.20.1 and 7.20.0
![Screen Shot 2020-07-05 at 18 13 31](https://user-images.githubusercontent.com/3503252/86535752-558b3580-beeb-11ea-99a9-1c6f5cbeef56.png)

**Comment by goooseman** [comment]
And it looks like the problem is already solved in the upstream: https://github.com/yannickcr/eslint-plugin-react/commit/4ee6f8e1ff15f89596e6c0249a21275a92052e58

**Comment by ljharb** [comment]
@goooseman that's included in v7.20.3. Have you tried that version?

**Comment by goooseman** [comment]
Yep, I've tried. And it was throwing the error, if Typescript interface has been used.

**Comment by ljharb** [comment]
@goooseman could you provide a reduced test case that's incorrect in the latest version?

**Comment by Multiply** [comment]
Using `eslint-plugin-react@7.20.3` we are also seeing the issue.

Example:

```
interface GenericProps {
  onClose: () => void
}

interface ImplementationProps extends GenericProps {
  onClick: () => void
}

export const Implementation: FC<ImplementationProps> = (
  {
    onClick,
    onClose,
  }: ImplementationProps
) => (
  // ... implementation here
)
```

It'll complain about the prop from the extended interface `onClose` missing in props validation.

**Comment by einsiol** [comment]
> By downgrading back to 7.20.0 (without downgrading any of libraries deps) the problem is resolved, so I believe the problem appeared somewhere between 7.20.1 and 7.20.0
> ![Screen Shot 2020-07-05 at 18 13 31](https://user-images.githubusercontent.com/3503252/86535752-558b3580-beeb-11ea-99a9-1c6f5cbeef56.png)

The issue I was having was on 7.20.0. I tried to update to 7.20.3, latest version but made no difference for the problem I mentioned above.

**Comment by hank121314** [comment]
For those who encounter this issue.
If you don't mind, could you please help me to test the extends feature in my develop branch
https://github.com/hank121314/eslint-plugin-react/tree/develop
It might resolve this issue.

**Comment by StallionV** [comment]
Reason: Seems like if you have used the word "props" in your objects this error gets triggered
e.g.  {name: "", props: {..} }
Those looking for a quick fix.
Reverting version back to : `7.16.0`
and deleting your package-lock.json, node modules directory and then doing `npm install` fixed this for us

But this needs an immediate fix to stop builds from breaking.

**Comment by Cretezy** [comment]
The upgrade from 7.20.0 -> 7.20.1 seems to be causing the issue for me.

**Comment by aaronlna** [comment]
> The upgrade from 7.20.0 -> 7.20.1 seems to be causing the issue for me.

Yup. Downgrading back to 7.20.0 resolved this for me.

**Comment by kylemh** [comment]
I wonder if this was also fixed in 7.20.4, like [here](https://github.com/yannickcr/eslint-plugin-react/issues/2719)

**Comment by damiangreen** [comment]
do we know which version this will be released in?

**Comment by ljharb** [comment]
@damiangreen if you click on the commit that closed the PR, you'll see it's not released in anything yet. So, the answer to your question is "the next release".

**Comment by S-Amin** [comment]
I just tested the version `7.20.6` and still the same error.
this is my package dependency 

```
        "eslint": "^7.7.0",
        "eslint-config-airbnb-typescript": "^9.0.0",
        "eslint-config-prettier": "^6.11.0",
        "eslint-plugin-import": "^2.22.0",
        "eslint-plugin-jsx-a11y": "^6.3.1",
        "eslint-plugin-prettier": "^3.1.4",
        "eslint-plugin-react": "^7.20.6",
        "eslint-plugin-react-hooks": "^4.0.8",

```

this is the `eslint` consfig file 
```

{
    "env": {
        "browser": true,
        "es2020": true
    },
    "extends": [
        "airbnb-typescript",
        // "eslint:recommended",
        // "plugin:react/recommended",
        "plugin:@typescript-eslint/recommended",
        "prettier",
        "prettier/react",
        "prettier/@typescript-eslint",
        "plugin:prettier/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "project": "./tsconfig.json",
        "ecmaFeatures": {
            "jsx": true
        },
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": ["react", "@typescript-eslint"],
    "rules": {
        "prettier/prettier": ["error"],
        "react/react-in-jsx-scope": "off",
        "import/prefer-default-export": "warn",
        "no-underscore-dangle": "off",
        "no-nested-ternary": "off",
        "@typescript-eslint/no-empty-interface": "warn",
        "@typescript-eslint/naming-convention": [
            "error",
            {
                "selector": "interface",
                "format": ["camelCase", "UPPER_CASE", "PascalCase"]
            }
        ]
    }
}

```

and there are multiple components and `eslint` threw this error `... is missing in props validation `

**Comment by ljharb** [comment]
@S-Amin thanks; could you file a new issue?

**Comment by chidioguejiofor** [comment]
@ljharb upgrading to the latest version `7.21.1` fixes this

**Comment by maxie7** [comment]
version 7.21.5 >> the same issue && the same error

**Comment by ljharb** [comment]
@maxie7 can you file a new issue for that?

**Comment by maxie7** [comment]
The problem was solved by installing "prop-types" v15.7.2 for React v17.0.1. So, it's not a bug, I think. Good luck with eslint plugin, it's awesome!

**Comment by aximusnl** [comment]
I'm at version `7.21.5`. And this problem still exists. But only when the interface extending from is imported:

This is OK
```
interface PropsBase {
	foo: string;
}

interface Props extends PropsBase {
	bar: string;
}

const MyComponent = ({ foo, bar}: Props): JSX.Element => {
	return (
		<div>
			{foo} {bar}
		</div>
	)
}

export {
	MyComponent,
}
```

This shows an error: 'foo' is missing in props validation

```
interface PropsBase {
	foo: string;
}
```

```
import { PropsBase } from './PropsBase';

interface Props extends PropsBase {
	bar: string;
}

const MyComponent = ({ foo, bar}: Props): JSX.Element => {
	return (
		<div>
			{foo} {bar}
		</div>
	)
}

export {
	MyComponent,
}
```

**Comment by Siphenathi-Nathi** [comment]
For me upgrading eslint-plugin-react to the latest version 7.21.5 fixed this

**Comment by thayllachristine** [comment]
in `v7.22.0` this problem continues! I did downgrade to cited versions, but the issue persists. :cry:

**Comment by petrunov** [comment]
This issue is reported almost a year ago ?

**Comment by sambhav-gore** [comment]
We are facing the same issue. with `v7.22.0`

**Comment by ljharb** [comment]
@sambhav-gore please file a new issue?

**Comment by zakur777** [comment]
we have the same problem with version 7.22, I tried to downgrade and I had no results either.

**Comment by ljharb** [comment]
@zakur777 just like the last three comments; please file a new issue?

**Comment by Tordek947** [comment]
Encounter the same issue using version 7.23.1

**Comment by DamengRandom** [comment]
> By downgrading back to 7.20.0 (without downgrading any of libraries deps) the problem is resolved, so I believe the problem appeared somewhere between 7.20.1 and 7.20.0
> ![Screen Shot 2020-07-05 at 18 13 31](https://user-images.githubusercontent.com/3503252/86535752-558b3580-beeb-11ea-99a9-1c6f5cbeef56.png)

not working for my case ...

**Comment by ljharb** [comment]
@DamengRandom then please file a new issue?

**Comment by DamengRandom** [comment]
Hi @ljharb,

 Would you know how to fix this issue without disable the eslint rule? I really wish this error can be fixed: (FYI, I am using typescript and material-ui)
 
![image](https://user-images.githubusercontent.com/18480984/125727734-e84e7224-ae57-4f7a-acab-96abbb6f1aec.png)

**Comment by DamengRandom** [comment]
Hi @ljharb, its really hard one for me now, but I use an alternative solution to resolve this issue:

which is instead of using arrow function, we use normal function, something like this:

`function DrawerList({ toggleDrawer }: DrawerListInterface): React.ReactElement { ... }`

NOT use this: 

const DrawerList: React.FC<DrawerListInterface> = ({ toggleDrawer }) => { ... }

However, I still want to typescript to fix this issue (at least if my codebase is using typescript, this rule does not require, which should only be available on es6, right?) Or would you mind to offer me some feedback?

**Comment by SalahAdDin** [comment]
> Hi @ljharb, its really hard one for me now, but I use an alternative solution to resolve this issue:
> 
> which is instead of using arrow function, we use normal function, something like this:
> 
> `function DrawerList({ toggleDrawer }: DrawerListInterface): React.ReactElement { ... }`
> 
> NOT use this:
> 
> `const DrawerList: React.FC<DrawerListInterface> = ({ toggleDrawer }) => { ... }`
> 
> However, I still want to typescript to fix this issue (at least if my codebase is using typescript, this rule does not require, which should only be available on es6, right?) Or would you mind to offer me some feedback?

It seems there is no any fix to this problem.
